raiden_contracts/tests/test_channel_open.py

```diff
@@ -96,6 +96,53 @@ def test_open_channel_state(token_network, get_accounts):
     assert B_nonce == 0
 
 
+def test_reopen_channel(
+        web3,
+        token_network,
+        get_accounts,
+):
+    (A, B) = get_accounts(2)
+    settle_timeout = TEST_SETTLE_TIMEOUT_MIN
+    locksroot = fake_bytes(32)
+
+    token_network.functions.openChannel(A, B, settle_timeout).transact()
+
+    # Opening twice fails
+    with pytest.raises(TransactionFailed):
+        token_network.functions.openChannel(A, B, settle_timeout).transact()
+
+    # Close channel
+    token_network.functions.closeChannel(
+        B,
+        locksroot,
+        0,
+        fake_bytes(32),
+        fake_bytes(64),
+    ).transact({'from': A})
+
+    # Reopen Channel before settlement fails
+    with pytest.raises(TransactionFailed):
+        token_network.functions.openChannel(A, B, settle_timeout).transact()
+
+    # Settlement window must be over before settling the channel
+    web3.testing.mine(settle_timeout)
+
+    # Settle channel
+    token_network.functions.settleChannel(
+        A,
+        0,
+        0,
+        locksroot,
+        B,
+        0,
+        0,
+        locksroot,
+    ).transact({'from': A})
+
+    # Reopening the channel should work iff channel is settled
+    token_network.functions.openChannel(A, B, settle_timeout).transact()
+
```

Comment:
Please also `check channel state to make sure we don't have lingering balance proofs from the previous one)`, as I mentioned here https://github.com/raiden-network/raiden-contracts/issues/149#issuecomment-400365496